File: app/src/main/java/remix/myplayer/request/SimpleUriRequest.java
package remix.myplayer.request;

import android.annotation.SuppressLint;
import android.net.Uri;
import android.support.annotation.NonNull;
import io.reactivex.disposables.Disposable;
import remix.myplayer.request.network.RxUtil;

public abstract class SimpleUriRequest extends ImageUriRequest<Uri> {

  private UriRequest mRequest;

  public SimpleUriRequest(@NonNull UriRequest request) {
    super();
    mRequest = request;
  }

  @SuppressLint("CheckResult")
  @Override
  public Disposable load() {
    return getCoverObservable(mRequest)
        .compose(RxUtil.applyScheduler())
        .subscribe(s -> onSuccess(Uri.parse(s)),
            throwable -> onError(throwable.toString()));
  }
}

File: app/src/main/java/remix/myplayer/request/SimpleUriRequest.kt
package remix.myplayer.request

import android.annotation.SuppressLint
import io.reactivex.disposables.Disposable

@Deprecated("replace with glide")
abstract class SimpleUriRequest(private val request: UriRequest) : ImageUriRequest<String>() {

  @SuppressLint("CheckResult")
  override fun load(): Disposable {
    return getCoverObservable(request)
        .subscribe({ s -> onSuccess(s) }, { throwable -> onError(throwable) })
  }


}
